drivers/usb/udc/udc_stm32.c

```diff
@@ -62,6 +62,10 @@ LOG_MODULE_REGISTER(udc_stm32, CONFIG_UDC_DRIVER_LOG_LEVEL);
 #else
 #define UDC_STM32_FULL_SPEED             USB_OTG_SPEED_FULL
 #endif
+/* Definitions aligned with HAL macros for PHY type selection */
+#define USB_OTG_HS_PHY_ULPI	1U
+#define USB_FS_EMBEDDED_PHY	2U
+#define USB_OTG_HS_EMBEDDED_PHY	3U
```

Comment:
Sorry, maybe I missed the explanations for these. But these macros are not properly prefixed and USB_OTG_HS_EMBEDDED_PHY is already defined in the [HAL](https://github.com/zephyrproject-rtos/hal_stm32/blob/main/stm32cube/stm32f7xx/drivers/include/stm32f7xx_ll_usb.h#L279-L281), would it not clash at some point? Also, I see two other values defined there, why cannot they be used here in the UDC driver?

Reply:
>USB_OTG_HS_EMBEDDED_PHY is already defined in the [HAL](https://github.com/zephyrproject-rtos/hal_stm32/blob/main/stm32cube/stm32f7xx/drivers/include/stm32f7xx_ll_usb.h#L279-L281), would it not clash at some point? Also, I see two other values defined there, why cannot they be used here in the UDC driver?

There is one question that remains unanswered. Why cannot the macros from the [HAL](https://github.com/zephyrproject-rtos/hal_stm32/blob/main/stm32cube/stm32f7xx/drivers/include/stm32f7xx_ll_usb.h#L279-L281) be used directly in the UDC driver?